SendLog.py

```diff
@@ -126,90 +220,576 @@ def event_stream(xcos_file_id):
+            print(line,end = '')
+
+
+        # Changing the file extension from xml to xcos
+        base_filename = os.path.splitext(temp_file_xml_name)[0]
+        os.rename(temp_file_xml_name, base_filename + ".xcos")
+        source_folder = os.getcwd()
+        destination_folder = source_folder + "/uploads/"
+
+
+        # Move the xcos file to uploads directory
+        folder_file_content = filter(os.path.isfile, os.listdir( os.curdir ) )
+        for file in folder_file_content:
+            if file.endswith(".xcos"):
+                shutil.copy(file, destination_folder)
+                subprocess.Popen(["rm",file])# After moving to the required folder deleting the xcos file.
+        temp_file_xml_name = base_filename + ".xcos"
+        xcos_file_list.append(temp_file_xml_name)
+        workspace_list.append(workspace_counter)
+        return str(client_id)
+    else:
+        return "error"
+
+@app.route('/UpdateTKfile', methods=['POST'])    
+def UpdateTKfile():		# '=> Dattatreya <='  function which makes the initialazation and updation of the files with obtained new value
 	# Get the file
 	file = request.files['file']
+
 	# Check if the file is not null
 	if file:
-		# Make the filename safe, remove unsupported chars
-		client_id = len(xcos_file_list)
-		filename = str(client_id)+".xcos"
-		# Move the file form the temporal folder to
-		# the upload folder we setup
-		file.save(os.path.join(app.config['UPLOAD_FOLDER'], filename))
-		xcos_file_list.append(filename)
-		return str(client_id)
+		filename1 = Details.uid+'_val.txt'
+		file.save(os.path.join(app.config['VALUES_FOLDER'], filename1)) # saves the file in values folder
+		n = findFile()
+		if n==0:
+		   getDetailsThread() # starts the thread
+
+		elif n==1:
+		   stopDetailsThread() # stops the thread
+		
+		return ""
 	else:
 		return "error"
+@app.route('/importXcos')   # '=> Dattatreya <=' function to take the request data when user click the url link (view example)
+def importXcos():
+
+	
+	Details.names[Details.uid+"eid"] = request.args.get('eid')	# get the eid from the url
+	
+	with open("webapp/xcos/"+str(Details.names[Details.uid+"eid"])+".xcos") as inf:
+		Details.names[Details.uid+"xcosContents"] = inf.read()	# read the xcos file contents
+	import bs4
+	with open("webapp/index.html") as inf:
+    		txt = inf.read()
+    		Details.names[Details.uid+"soup"] = bs4.BeautifulSoup(txt,'html.parser')	# load soup to edit the html file
+	
+	Details.names[Details.uid+"pidlist"] = Details.names[Details.uid+"soup"].findAll("p")
+	Details.names[Details.uid+"innerhtml_1"] = Details.names[Details.uid+"pidlist"][11].text
+	temp = Details.names[Details.uid+"soup"].find(text=Details.names[Details.uid+"innerhtml_1"]).replaceWith(Details.names[Details.uid+"xcosContents"])	
+
+	with open("webapp/index.html", "w") as file:
+    		file.write(str(Details.names[Details.uid+"soup"]))				# edit and save the changes in html file
+
+	import webbrowser
```

Comment:
It seems that you are trying to dynamically render something in a simulated browser. Please avoid, try to use a new URL and template.